### PR TITLE
Ensure correct claim logic is applied, clarify cases.

### PR DESCRIPTION
**Pull request by cfelde**
Fix for #7

### CODE REVIEW

**Comment by cfelde** [comment]
@mohamedelshami care to take a quick look at my changes?

File: contracts/AKAP.sol [hunk 1]
```diff
@@ -56,24 +56,36 @@ contract AKAP is ERC721Full {
...
+        bool isParentOwner = parentId == 0x0 || _isApprovedOrOwner(_msgSender(), parentId);
+
         if (_exists(nodeId) && _isApprovedOrOwner(_msgSender(), nodeId)) {
             // Caller is current owner/approved, extend lease..
             nodes[nodeId].expiry = now + 52 weeks;
 
             return 1;
-        } else if (!_exists(nodeId)) {
+        } else if (!_exists(nodeId) && isParentOwner) {
             // Node does not exist, allocate to caller..
```
Comment (mohamedelshami) [inline contracts/AKAP.sol:83]: We maybe need to reconsider moving this to a a separate function. The special case of parentId = 0x0, can also be seen as base case nodeId is assigned and has no owner.

File: contracts/AKAP.sol [hunk 2]
```diff
@@ -105,6 +117,10 @@ contract AKAP is ERC721Full {
         return nodes[nodeId].nodeBody;
     }
 
+    function expireNode(uint nodeId) external onlyApproved(nodeId) {
+        nodes[nodeId].expiry = now;
+    }
+
```
Comment (mohamedelshami) [inline contracts/AKAP.sol:123]: Is this away to relinquish the ownership without doing a transfer?

**Review by mohamedelshami — APPROVED** [review]

File: contracts/AKAP.sol (hunk 2 above)
Comment (cfelde) [reply contracts/AKAP.sol:123]: Ownership doesn't change, but it makes the node available for someone else to claim.

File: contracts/AKAP.sol (hunk 1 above)
Comment (cfelde) [reply contracts/AKAP.sol:83]: Not clear on what should be considered moving? The isParentOwner check?

Comment (mohamedelshami) [reply contracts/AKAP.sol:83]: The whole case 2 block, and restrict claim to onlyApproved and existing. Still need to account for node expired case though. 

function claimNew(parentId, label) external NonExisting {
   // Case 2 here
}

Comment (cfelde) [reply contracts/AKAP.sol:83]: Not keen on having multiple claim functions as that would force the caller to apply logic prior to making a call, and that becomes tricky in an asynchronous/eventually consistent world.

Better to encapsulate such logic within the claim function, that way the caller knows they can simply use that one function without thinking about existing state.

**Pull request merged**